### PR TITLE
skip update taxes for lines with deleted variant

### PR DESCRIPTION
**Pull request by jakubkuc**
I want to merge this change because it fixes problem with updating orders with deleted variants

# Impact

* [ ] New migrations
* [ ] New/Updated API fields or mutations
* [ ] Deprecated API fields or mutations
* [ ] Removed API types, fields, or mutations
* [ ] Documentation needs to be updated

# Pull Request Checklist

* [ ] Privileged queries and mutations are guarded by proper permission checks
* [ ] Database queries are optimized and the number of queries is constant
* [ ] Database migration files are up to date
* [ ] The changes are tested
* [ ] GraphQL schema and type definitions are up to date
* [ ] Changes are mentioned in the changelog

### CODE REVIEW

**Comment by patrys** [comment]
Task linked: [SALEOR-5477 Changing address doesn't work in orders with deleted variants](https://app.clickup.com/t/2549495/SALEOR-5477)

File: saleor/order/utils.py [hunk 1]
```diff
@@ -195,6 +195,8 @@ def update_taxes_for_order_line(
     line: "OrderLine", order: "Order", manager, tax_included
 ):
     variant = line.variant
+    if not variant:
+        return
```
Comment (IKarbowiak) [inline saleor/order/utils.py:199]: Could you add a test for the situation which is fixed?

Comment (jakubkuc) [reply saleor/order/utils.py:199]: added

**Review by IKarbowiak — APPROVED** [review]

File: saleor/order/tests/test_order_utils.py [hunk 2]
```diff
@@ -326,3 +328,41 @@ def test_add_variant_to_order(order, customer_user, variant, site_settings):
...
+
+def test_update_taxes_for_order_line_deleted_variant(order_with_lines):
+    # given
+    order_line = order_with_lines.lines.first()
+    order_line_unchanged_copy = copy.deepcopy(order_line)
+    unit_discount_amount = Decimal("2.00")
+
+    unit_price = TaxedMoney(net=Money("10.23", "USD"), gross=Money("15.80", "USD"))
+    total_price = TaxedMoney(net=Money("30.34", "USD"), gross=Money("36.49", "USD"))
+    tax_rate = Decimal("0.23")
```
Comment (maarcingebala) [inline saleor/order/tests/test_order_utils.py:341]: Where are these values coming from, are they hardcoded? Will the test pass if we make an unrelated change in `order_with_lines` fixture and it would return different values?

Comment (jakubkuc) [reply saleor/order/tests/test_order_utils.py:341]: Those values are different from values in the fixture (they are copied from other tests from `test_order.utils.py`). `update_taxes_for_order_line` method should not update prices in line with those values. So changing anything in fixture should not interfere with the test. The test only checks if no changes in prices were made by `update_taxes_for_order_line` in line when variant is `None`.

Comment (maarcingebala) [reply saleor/order/tests/test_order_utils.py:341]: 👍

**Review by maarcingebala — APPROVED** [review]
